src/sparql/TripleBuilder.ts

```diff
@@ -19,6 +20,16 @@ export default class TripleBuilder {
 		};
 	}
 
+	private checkValueType( value: string | UnitValue ): string | number {
+		if ( typeof value === 'string' ) {
+			return value;
+		}
+		if ( 'value' in value ) { // if value is UnitValue
+			return value.value; // TODO: multiply by unit, define how to manage it
```

Comment:
Are we doing any unit conversion at some point??

Reply:
Yes, as it was [discussed and documented](https://phabricator.wikimedia.org/T276938), we will have two different queries depending on whether this is a unit or not.

In this specific case here I would suggest to throw an error after the first `if`, instead of returning a number. That is because that number will still lead to a wrong query and so it is better to fail explicitly and loudly.

The proper implementation will be done in the course of [T276938](https://phabricator.wikimedia.org/T276938).

Reply:
Sounds good!